### PR TITLE
More thorough discussion of sequencing solutions.

### PR DESCRIPTION
**Pull request by Bren2010**
This PR changes the "client / server ordering" distinction to more appropriate "CP / AP ordering".

### CODE REVIEW

File: draft-ietf-mls-protocol.md [hunk 1]
```diff
@@ -4352,71 +4352,75 @@ current state of the group.  In practice, however, there is a risk
...
+classes of distributed system:
+
+* Consistent and Partition-tolerant, or CP, systems can provide a globally
+  consistent view of data but may stop working if there are network issues.
+* Available and Partition-tolerant, or AP, systems continue working despite
+  network issues but may return different views of data to different users.
+
+Strategies for sequencing messages in CP and AP systems are described in the
+next two sections. Implementations MUST only update their cryptographic state
+when they have a reasonable degree of certainty that their Commit was accepted.
```
Comment (rohan-wire) [inline draft-ietf-mls-protocol.md:4366]: This whole discussion belongs in the architecture document and not the protocol document.

File: draft-ietf-mls-protocol.md [hunk 2]
```diff
@@ -4352,71 +4352,75 @@ current state of the group.  In practice, however, there is a risk
...
-
-Regardless of how messages are kept in sequence, implementations
-MUST only update their cryptographic state when valid Commit
-messages are received.
-Generation of Commit messages MUST NOT modify a client's state, since the
-endpoint doesn't know at that time whether the changes implied by
-the Commit message will succeed or not.
-
-## Server-Enforced Ordering
+## CP Ordering
```
Comment (rohan-wire) [inline draft-ietf-mls-protocol.md:4379]: No thank you. Please do not a use a relatively rare acronym when the reading audience will understand "Server-Enforced Ordering" just fine.

File: draft-ietf-mls-protocol.md [hunk 3]
```diff
@@ -4352,71 +4352,75 @@ current state of the group.  In practice, however, there is a risk
...
+by the Delivery Service and used for tie-breaking. The counter would start at 0
+and be incremented for each subsequent message. If two group members send a
+message with the same counter, or two Commits with the same `epoch`, the first
+one to arrive would be accepted and the second one would be rejected. The
+rejected message would either be dropped by the client or resent later.
+
+The counter approach would be done to apply ordering to all messages, while the
+`epoch` approach would be done to only ensure an order on Commits.
+
+## AP Ordering
```
Comment (rohan-wire) [inline draft-ietf-mls-protocol.md:4397]: Likewise, using "AP Ordering" is going this document less clear.

File: draft-ietf-mls-protocol.md [hunk 4]
```diff
@@ -4352,71 +4352,75 @@ current state of the group.  In practice, however, there is a risk
...
+   clients use a deterministic tie-breaking policy to decide if they should
+   continue using the Commit they originally accepted or revert and use the
+   later one. Note that any copies of previous or forked group states MUST be
+   deleted within a reasonable amount of time to ensure the protocol provides
+   forward-secrecy.
+
+In the event of a network partition, a subset of members may be isolated from
+the rest of the group long enough that the mechanisms above no longer work. This
+can only be solved by sending a ReInit proposal to both groups, possibly with an
+external sender type, and recreating the group to contain all members again.
```
Comment (rohan-wire) [inline draft-ietf-mls-protocol.md:4423]: I propose we punt all this to the architecture document

**Review by rohan-wire — CHANGES_REQUESTED** [review]

**Comment by bifurcation** [comment]
Interim 2022-05-26:

* General feeling that the details here should go in the architecture document
* @Bren2010 to make a new PR stripping the sequencing section down to minimal requirements, arch PR to add these details.

**Pull request closed without merging**